denizen_scripts/hub1/repo-link/Adriftus_Moderator_Panel/tools/mod_report.dsc

```diff
@@ -0,0 +1,30 @@
+mod_report_command:
+  type: command
+  debug: true
+  name: report
+  description: Report a player for chat & in-game behaviour.
+  usage: /report [username] [reason]
+  tab completions:
+    - define players <server.online_players.parse[name].exclude[<player.name>]>
+    - if <context.args.is_empty>:
+      - determine <[players]>
+    - else if <context.args.size> == 1 && <context.raw_args.ends_with[<&sp>].not>:
+      - determine <[players].filter[starts_with[<context.args.first>]]>
+  script:
+    - if <context.args.is_empty>:
+      - narrate "<&c>Adriftus Reporter"
+      - narrate "<&6>/report [username] [reason]"
+    - else if <server.match_offline_player[<context.args.first>]||null> != null:
+      - define target <server.match_offline_player[<context.args.first>]>
+      - if <[target]> != <player>:
+        - if <context.args.get[2]||null> != null:
+          - define reason <context.args.get[2].to[<context.args.size>].space_separated>
+          - run mod_notify_report def:<player>|<[target]>|<[reason]>|<bungee.server||Server>
```

Comment:
In the case that a player types an unfortunate combination of raw object notation or a pipe, this would break this definition.
```suggestion
          - run mod_notify_report def:<list[<player>|<[target]>].include_single[<[reason]>].include[<bungee.server||Server>]>
```